volume/drivers/nfs/nfs.go

```diff
@@ -285,31 +292,121 @@ func (d *driver) Create(
 		}
 	}
 
-	f, err := os.Create(path.Join(volPathParent, volumeID+nfsBlockFile))
-	if err != nil {
-		logrus.Println(err)
-		return "", err
-	}
-	defer f.Close()
+	// Create volume
+	var v *api.Volume
+	if spec.GetShared() {
+		// File based
+		v = common.NewVolume(
+			volumeID,
+			api.FSType_FS_TYPE_NFS,
+			locator,
+			source,
+			spec,
+		)
+
+		// Create a directory on the NFS server with this UUID.
+		err := os.MkdirAll(volPath, 0744)
+		if err != nil {
+			logrus.Println(err)
+			return "", err
+		}
 
-	if err := f.Truncate(int64(spec.Size)); err != nil {
-		logrus.Println(err)
-		return "", err
-	}
+		// Setup volume
+		v.State = api.VolumeState_VOLUME_STATE_PENDING
+		if err := d.CreateVol(v); err != nil {
+			return "", err
+		}
 
-	v := common.NewVolume(
-		volumeID,
-		api.FSType_FS_TYPE_NFS,
-		locator,
-		source,
-		spec,
-	)
-	v.DevicePath = path.Join(volPathParent, volumeID+nfsBlockFile)
+		// Check for cloning
+		if source != nil && len(source.GetParent()) != 0 {
+			// Need to clone
+			if err := d.clone(volumeID, source.GetParent()); err != nil {
+				d.Delete(v.GetId())
+				return "", err
+			}
+		}
 
-	if err := d.CreateVol(v); err != nil {
-		return "", err
+		// Set to ready
+		v.State = api.VolumeState_VOLUME_STATE_AVAILABLE
+		if err := d.UpdateVol(v); err != nil {
+			d.Delete(v.GetId())
```

Comment:
if this fails - are we okay with ignoring the error? Maybe log that cleanup failed?